gdnative-core/src/core_types/string.rs

```diff
@@ -260,6 +260,53 @@ impl std::hash::Hash for GodotString {
     }
 }
 
+impl Add<&GodotString> for &GodotString {
+    type Output = GodotString;
+    #[inline]
+    fn add(self, other: &GodotString) -> GodotString {
+        GodotString::from_sys(unsafe { (get_api().godot_string_operator_plus)(&self.0, &other.0) })
+    }
+}
+
+impl<S> Add<S> for &GodotString 
+where 
+    S: AsRef<str>,
+{
+    type Output = GodotString;
+    #[inline]
+    fn add(self, other: S) -> GodotString {
+        self.add(&GodotString::from_str(other))
+    }
+}
+
+impl AddAssign<&GodotString> for GodotString {
+    #[inline]
+    fn add_assign(&mut self, other: &Self) {
+        let mut new_string = &*self + other;
+        new_string.0 = std::mem::replace(&mut self.0, new_string.0);
```

Comment:
Maybe just replace the whole string here, since that is simpler and should make no run-time difference?

```rust
std::mem::replace(self, new_string);
```